src/components/QueryRecordings/SelectDevice.vue

```diff
@@ -29,66 +33,67 @@ export default {
       }
     },
     ...mapState({
       fetched: state => state.Devices.fetched,
       devices: state =>
         state.Devices.devices.map(device => {
           return {
-            id: device.id,
+            id: "D" + device.id,
             name: device.devicename
           };
         }),
       groups: state =>
         state.Groups.groups.map(group => {
           return {
-            id: "group" + group.id,
+            id: "G" + group.id,
             name: group.groupname + " (group)",
             devices: group.Devices
           };
         })
     }),
     selectedValues: function() {
-      // Allow the devices to just be defined by ids if we haven't already fetched
-      // the device labels etc. by the time the parent component initialises this one,
-      // supporting the use-case where we're re-populating this from a shared url.
-
-      // Gather a list of devices that belong to selected groups.
-      const groupDevices = this.value
-        .filter(id => typeof id === "string")
-        .map(groupId => this.groups.find(({ id }) => groupId === id))
-        .filter(item => item !== undefined)
-        .reduce((acc, group) => [...acc, ...group.devices], []);
-
-      return this.value
-        .map(device => {
-          if (typeof device === "number") {
-            // This is a device id:
-            return this.devices.find(({ id }) => device === id);
-          } else if (typeof device === "string") {
-            // This is a group id:
-            return this.groups.find(({ id }) => device === id);
-          } else {
-            return device;
-          }
-        })
-        .filter(
-          device =>
-            device !== undefined &&
-            // Remove items that belong to any selected groups:
-            groupDevices.find(groupDevice => groupDevice.id === device.id) ===
-              undefined
-        );
+      const selectedDs = this.selectedDevices.map(deviceId =>
+        this.devices.find(({ id }) => "D" + deviceId === id)
+      );
+      const selectedGs = this.selectedGroups.map(groupId =>
+        this.groups.find(({ id }) => "G" + groupId === id)
+      );
+      return [...selectedDs, ...selectedGs];
     },
     options: function() {
       return this.devices.concat(this.groups);
     }
   },
+  methods: {
+    updateSelected(selectedObjects) {
+      const devices = this.getIdsWithPrefix(selectedObjects, "D");
+      const groups = this.getIdsWithPrefix(selectedObjects, "G");
+      const updatedSelection = {
+        devices: devices,
+        groups: groups
+      };
+      // this causes the v-model in the parent component to get updated
+      this.$emit("update-device-selection", updatedSelection);
+    },
+
+    getIdsWithPrefix(objects, prefix) {
+      const prefixed = objects.filter(
+        item => item.id.substring(0, prefix.length) === prefix
+      );
+      return prefixed.map(item =>
+        parseInt(item.id.substring(prefix.length, item.id.length))
```

Comment:
These days you can safely use `item.id.startsWith(prefix)` if you want: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith